api/v1beta2/foundationdbcluster_types.go

```diff
@@ -202,6 +202,11 @@ type FoundationDBClusterSpec struct {
 	// +kubebuilder:default:=false
 	ReplaceInstancesWhenResourcesChange *bool `json:"replaceInstancesWhenResourcesChange,omitempty"`
 
+	// RemoteStorage defines if the cluster is using remote storage. This is used to determine
+	// how the operator should handle pod replacements.
```

Comment:
Can we actually document the current limitations? e.g. this will only affect node selectors. We should also note here that some cases might require to set this value to false, even if remote storage is used. One example would be if a user wants to migrate the cluster from one AZ to another AZ in AWS (as EBS volumes are limited to an AZ).